app/jobs/notify_expired_rents_job.rb

```diff
@@ -0,0 +1,17 @@
+class NotifyExpiredRentsJob
+  include Sidekiq::Worker
+
+  def perform
+    Rent.ending_today.find_in_batches do |rent_batch|
+      rent_batch.each do |rent|
+        notify_expried_rent(rent)
+      end
+    end
+  end
+
+  private
+
+  def notify_expried_rent(rent)
+    RentMailer.with(rent: rent).expired_rent_email.deliver_now
```

Comment:
Why `deliver_now` here? You will lock the thread waiting each mail will be send.

Reply:
So, try to send only the `id` not all the object (I know we do not resolve this question, my bad, but we will stay like that to avoid possible problems)